Provider/src/main/java/com/spotify/confidence/ConfidenceFlagEvaluation.kt

```diff
@@ -0,0 +1,134 @@
+package com.spotify.confidence
+
+import com.spotify.confidence.client.ResolveReason
+
+internal fun <T> FlagResolution?.getEvaluation(
+    flag: String,
+    defaultValue: T,
+    context: Map<String, ConfidenceValue>,
+    applyFlag: (String, String) -> Unit = { _, _ -> }
+): Evaluation<T> {
+    val parsedKey = FlagKey(flag)
+    if (this == null) {
+        return Evaluation(
+            value = defaultValue,
+            reason = ResolveReason.ERROR,
+            errorCode = ErrorCode.PROVIDER_NOT_READY
+        )
+    }
+    requireNotNull(this)
+    val resolvedFlag = this.flags.firstOrNull { it.flag == parsedKey.flagName }
+        ?: throw FlagNotFoundError(
+            "Could not find flag named: ${parsedKey.flagName}"
+        )
+
+    // TODO revisit where to apply flags
```

Comment:
Shall we address this TODO? From a quick glance it doesn't seem like the apply logic has been changed

Reply:
todo is removed.